01/implement-most-frequent-k-elements.instructor.md

```diff
@@ -0,0 +1,29 @@
+# Instructor: Implement Most Frequent K Elements
+
+An example of a working implementation:
+
+```python
+def top_k_frequent_elements(arr, k):
+    if len(arr) == 1: return [arr[0]]
+    
+    hash_table = {}
+    result = []
+
+    for num in arr:
+        if hash_table.get(num, False):
+            hash_table[num] += 1
+            
+        else:
+            hash_table[num] = 1
+            result.append(num)
+            
+    i = 0
+    while i < len(result) - 1:
```

Comment:
Both the outer and inner loops (i and j) both run to completion every time. Prefer using euther _both_ `while` loops or _both_ `for` loops. In this case, I'd probably go with `for`. I think the `while` had been traditionally preferred in older CS Fun-dedicated lessons to avoid making algorithms look too pythonic, but I think we're fine here.

Reply:
Alternatively, we could call sort(ed) to bring the time complexity down to O(n log n).

Reply:
Okay thank you! I didn't know if they were allowed to use `sort`, I'll replace what I have with that!

Reply:
If the problem isn't specifically to sort something (sorting here is only a portion of the approach) I wouldn't expect a student (or interview candidate) to implement a sort algorithm, especially when the sort provided in most languages is O(n log n), but a sort we'd implement on the spot is likely to be O(n^2).

Reply:
Thank you!